### ISSUE TITLE
Python 3.6.2

### ISSUE THREAD

**Issue by kuri65536** [kuri65536/python-for-android#138]

**Comment by kuri65536** [comment]
- Python 3.6.2. Release Date: 2017-07-17.
- compilation is done: 2017-08-30.

### fix points

- fix openssl/Makefile: $(dir ) -> $(subst $(dir)
- revert patch: work around press "y" -> remove patch
- can't compile Python/Modules/posix.c with NDK 10e: comment out Py_ASSERT
- platform string is empty? -> fix patch file name to /_sysconfigdata_m_linux_.py

### build environment changed
- OS: Bash On Windows (aka Ubuntu 16.04?)

**Comment by Crakkerjakked** [comment]
Every time I install, I get an error:

Error: java.io.FileNotFoundException:http://python-for-android.googlecode.com/hg/python3-alpha/VERSIONS

I've uninstalled the APK and reinstalled with no help. Still the same error. Clicking the Install button leads to "Installation Failed" on the bottom of the screen. 

Help?

**Comment by kuri65536** [comment]
use my apk: https://github.com/kuri65536/python-for-android/releases/download/r26/Python3ForAndroid-debug-r26.apk

but I forgot to upload r32 release, you will get an error cannot download r32 binaries.
please wait for minites.

**Comment by Joeyz321** [comment]
What causes a Python segmentation fault?

I installed python and python3 in my device, I attached it to /system/xbin and /system/lib, for the python I installed and it works fine in terminal emulator, but python3 does not work, it says "python segmentation fault". how to fix that? 

Thanks

**Comment by kuri65536** [comment]
Hi, Joeyz321.

I often use to debug `python -v` or `python -vv` option and see log if I face the seg-fault.

**Comment by Joeyz321** [comment]
![python-android](https://user-images.githubusercontent.com/32101857/30641629-20e17c4a-9e32-11e7-8402-7ed02d02e097.png)
![python-android-2](https://user-images.githubusercontent.com/32101857/30641631-23e7f9f0-9e32-11e7-8861-987219f25852.png)
![python-android-3](https://user-images.githubusercontent.com/32101857/30641648-2af7642e-9e32-11e7-8edf-4c2da038b04c.png)
![python3-android](https://user-images.githubusercontent.com/32101857/30641655-3122449a-9e32-11e7-9624-02aced959e95.png)

**Comment by kuri65536** [comment]
Python3 is newest [python3.6.2](https://github.com/kuri65536/python-for-android/releases/tag/r32) ?  

My phone can run python3.6.2 binary.
[standalone script](https://github.com/kuri65536/python-for-android/blob/master/sl4atools/python3.sh)
setup some environment variables to run python3, please refer it.

**Comment by Joeyz321** [comment]
I tried, but still the same, no change

**Comment by kuri65536** [comment]
I'm sad to hear that.
What your environment?: Android version, system (phone or board?)
My environment is Android 6, MediaTek X20 Phone.

**Comment by Joeyz321** [comment]
I use Samsung Galaxy Ace 3 GT-S7270 (logan).

ROM: LineageOS 11
Android : Kitkat 4.4.4
Chipset  : Broadcom BCM21664
CPU.       : Dual-core 1.0 GHz Cortex-A9

**Comment by kuri65536** [comment]
Thank you report.
I had mistake to patching files to build Python3.6.2 binary.
(some patches does not applied around [setlocale](https://github.com/kuri65536/python-for-android/blob/master/python3-alpha/patches/python-3.6.0-python-setlocale.patch) )

I found the my old phone and repreesnt problem.
wait for a minutes a replace the binary 

here is logcat output:
```
/PhoneStatusBar(11327): setLightsOn(true)
D/sl4a.ConsoleActivity$1:145( 4948): Connected to TerminalManager and found bridges.size=0
D/sl4a.ConsoleActivity$1:160( 4948): We couldnt find an existing bridge with id = 40561, so creating one now
D/sl4a.Relay:73( 4948): changing charset to UTF-8
I/InputMethodManagerService(11243): IME STATUS ON
I/DEBUG   (  420): [2017-09-24 08:19:00.270]
I/DEBUG   (  420): *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
I/DEBUG   (  420): Build fingerprint: 'lge/aka_open_hk/aka:4.4.2/KOT49I.H788n10d/H788n10d.1441603541:user/release-keys'
I/DEBUG   (  420): Revision: '6'
I/DEBUG   (  420): pid: 5243, tid: 5243, name: python3  >>> /data/user/0/com.googlecode.python3forandroid/files/python3/bin/python3 <<<
I/DEBUG   (  420): AM write failure (32 / Broken pipe)
I/DEBUG   (  420): signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 00000000
D/dalvikvm( 4948): GC_FOR_ALLOC freed 605K, 32% free 14735K/21468K, paused 31ms, total 31ms
I/DEBUG   (  420):     r0 00000000  r1 00000000  r2 400afe00  r3 00000000
I/DEBUG   (  420):     r4 beec46a4  r5 00000001  r6 00000000  r7 00000008
I/DEBUG   (  420):     r8 414a0020  r9 00000000  sl 00000000  fp 00000000
I/DEBUG   (  420):     ip 4032fbe0  sp beec4630  lr 4019fe6c  pc 40088a68  cpsr 60030030
I/DEBUG   (  420):     d0  0000000000000000  d1  0000000000000000
I/DEBUG   (  420):     d2  0000000000000000  d3  0000000000000000
I/DEBUG   (  420):     d4  0000000000000000  d5  0000000000000000
I/DEBUG   (  420):     d6  0000000000000000  d7  781bf50000000000
I/DEBUG   (  420):     d8  0000000000000000  d9  0000000000000000
I/DEBUG   (  420):     d10 0000000000000000  d11 0000000000000000
I/DEBUG   (  420):     d12 0000000000000000  d13 0000000000000000
I/DEBUG   (  420):     d14 0000000000000000  d15 0000000000000000
I/DEBUG   (  420):     d16 41de0798e1f8a3d7  d17 3f50624dd2f1a9fc
I/DEBUG   (  420):     d18 41a2ff859a000000  d19 0000000000000000
...
```

**Comment by Joeyz321** [comment]
Okay, I wait for the fix

**Comment by Joeyz321** [comment]
I want to ask one more thing.

If I want to install a python module, like 'Twisted'.
whether it can download directly from the official site and install it, or python module that must be compiled beforehand by you?

**Comment by kuri65536** [comment]
Had you asked me about Twisted module?
[I compiled Twisted](https://github.com/kuri65536/python-for-android/commit/592df7897196ddecf96ca4859dc939f3a5a5e88a)
 and upload to [release tab](https://github.com/kuri65536/python-for-android/releases/tag/py3mod-1.0.0)

But as you know, python3 does not work on Android4.4 now.
Please wait a minutes, I work on it.

**Comment by Joeyz321** [comment]
Yes, I am asking about python module.
does it work also in Python2.7 ?

**Comment by kuri65536** [comment]
I do not maintain C modules for Python2.
And modules for Python3 does not work with Python2.7.

If you want creating C modules,
you need the full build environment of python-for-android.
And also need to patch some workaround for Twisted.
These are a little difficult.

**Comment by Joeyz321** [comment]
Oh, okay I understand. That must be tough

**Comment by kuri65536** [comment]
@Joeyz321 
I confirmed Python3 with my Android 4.4 emulator.
Becareful, You can not set PYTHONHOME variable.
You may get 'can't initialize sys' error with PYTHONHOME variable.

This error was caused by https://github.com/rave-engine/python3-android/issues/7 (maybe)

**Comment by kuri65536** [comment]
I reopen https://github.com/kuri65536/python-for-android/issues/88
PYTHONHOME issue will be watched at #88

**Comment by Joeyz321** [comment]
It's been better than ever, but there's still a bit of a problem ...
![screenshot_2017-09-30-22-41-53](https://user-images.githubusercontent.com/32101857/31047225-c5e74a16-a630-11e7-8f28-de486bf4cee9.png)
![screenshot_2017-09-30-22-42-23](https://user-images.githubusercontent.com/32101857/31047226-cc96906a-a630-11e7-829f-5cbabc42f5e1.png)

**Comment by kuri65536** [comment]
Error was caused by you did not use Python3 installer apk.

- You need python3-extra.zip too.
- You  should maintain some environment variable see [python3.sh](https://github.com/kuri65536/python-for-android/blob/master/sl4atools/python3.sh)

Maybe your PYTHONPATH is not set collectly.
You can explain more detail and will get easy to find problem root.
detail: what you do, your environment, your commands.

cheer.

**Comment by Joeyz321** [comment]
Oh, Okay Thanks...

**Comment by kuri65536** [comment]
10/3 python 3.6.3 released, issue moved to #143
